clustered/common/src/main/java/org/ehcache/clustered/common/messages/EhcacheCodec.java

```diff
@@ -37,46 +34,30 @@
 
   @Override
   public byte[] encodeMessage(EhcacheEntityMessage message) {
-    return marshall(message);
+    if (message.getType() == EhcacheEntityMessage.Type.LIFECYCLE_OP) {
+      return LifeCycleOpCodec.encode((LifecycleMessage) message);
```

Comment:
I fail to see benefit in having static codecs though I have plenty of downsides coming to mind. Care to chime in?